src/arkouda_server.chpl

```diff
@@ -285,17 +288,18 @@ proc main() {
             }
 
             /*
-             * Declare the repTuple and binaryRepMsg variables, one of which is processed, if 
-             * applicable, and sent to sendRepMsg depending upon whether a string (repTuple)
-             * or bytes (binaryRepMsg) is to be returned.
+             * For messages that return a string repTuple is filled. For binary
+             * messages the message is sent directly to minimize copies.
              */
-            var binaryRepMsg: bytes;
             var repTuple: MsgTuple;
 
             select cmd
             {
-                when "array"             {repTuple = arrayMsg(cmd, payload, st);}
-                when "tondarray"         {binaryRepMsg = tondarrayMsg(cmd, args, st);}
+                when "array"             {repTuple = arrayMsg(cmd, args, payload, st);}
+                when "tondarray"         {
+                  var binaryRepMsg = tondarrayMsg(cmd, args, st);
+                  sendRepMsg(binaryRepMsg);
+                }
```

Comment:
Trivial/Stylistic: If you wanted to keep with the flow of the code here, it looks like you can make that a one-liner
```
sendRepMsg(tondarrayMsg(cmd, args, st));
```

Reply:
I have no strong preference here. I went with the current approach to make it stick out a little more since it is doing something different from every other control flow path, but I'd be fine switching to this if others prefer.